hack/xref-quadlet-docs

```diff
@@ -0,0 +1,261 @@
+
+###############################################################################
+# BEGIN boilerplate args checking, usage messages
+
+sub usage {
+    print  <<"END_USAGE";
+Usage: $ME [OPTIONS]
+
+$ME cross-checks quadlet documentation between the Go source[Go]
+and the man page[MD].
+
+ [Go]: $Go
+ [MD]: $Doc
+
+We check that:
+
+  * all keys in [Go] are documented in [MD]
+  * all keys in [MD] exist in [Go]
+    * any keys listed in [MD] tables also have a description block
+      and vice-versa
+  * all keys everywhere are in sorted order
+
+OPTIONS:
+
+  --help         display this message
+  --version      display program name and version
+END_USAGE
+
+    exit;
+}
+
+# Command-line options.  Note that this operates directly on @ARGV !
+our $debug   = 0;
+sub handle_opts {
+    use Getopt::Long;
+    GetOptions(
+        'debug!'     => \$debug,
+
+        help         => \&usage,
+        man          => \&man,
+        version      => sub { print "$ME version $VERSION\n"; exit 0 },
+    ) or die "Try `$ME --help' for help\n";
+}
+
+# END   boilerplate args checking, usage messages
+###############################################################################
+
+############################## CODE BEGINS HERE ###############################
+
+# The term is "modulino".
+__PACKAGE__->main()                                     unless caller();
+
+# Main code.
+sub main {
+    # Note that we operate directly on @ARGV, not on function parameters.
+    # This is deliberate: it's because Getopt::Long only operates on @ARGV
+    # and there's no clean way to make it use @_.
+    handle_opts();                      # will set package globals
+
+    # No command-line args
+    die "$ME: Too many arguments; try $ME --help\n"                 if @ARGV;
+
+    my $errs = 0;
+    $SIG{__WARN__} = sub {
+        print STDERR "@_";
+        ++$errs;
+    };
+
+    # Assume that Go source file has Truth
+    my $true_keys = read_go($Go);
+
+    # Read md file, compare against Truth
+    crossref_doc($Doc, $true_keys);
+
+    exit $errs;
+}
+
+
+#############
+#  read_go  #  Returns list of key strings found in quadlet.go
+#############
+sub read_go {
+    my $path = shift;
+    open my $fh, '<', $path
+        or die "$ME: Cannot read $path: $!\n";;
+
+    my @found;                          # List of key strings
+    my $last_constname;                 # Most recently seen const name
+
+    while (my $line = <$fh>) {
+        # Only interested in lines of the form   KeyFoo = "Foo"
+        if ($line =~ /^\s+Key(\S+)\s+=\s+"(\S+)"/) {
+            my ($constname, $keystring) = ($1, $2);
+
+            # const name *should* be the same as the string, except (sigh)
+            # there's an inexplicable block of KeyNetworkFoo = "Foo".
+            unless ($constname eq $keystring || $constname eq "Network$keystring") {
+                # There's also, sigh again, some preexisting mismatches
+                unless ($constname =~ /Remap[GU]ID/) {
```

Comment:
Barf. Since these are all undocumented, would it be possible to actually fix them?

Reply:
As I explained, these keys were deprected. I see that the code looks for `deprecated`. So, I guess the solution is to add this comment to these lines